### PR TITLE
digital root method

### PR DESCRIPTION
**Pull request by meg-1**
### Describe your change:

**added the 'digital_root' method to maths/sum_of_digits.py.
a digital root is: https://en.wikipedia.org/wiki/Digital_root**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x]  This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by poyea** [comment]
`FAILED arithmetic_analysis/in_static_equilibrium.py::arithmetic_analysis.in_static_equilibrium.polar_force` can be ignored. See https://github.com/TheAlgorithms/Python/issues/6170

**Comment by meg-1** [comment]
@Kush1101 could you please check out this pull request?

**Comment by meg-1** [comment]
@Kush1101 could you please check out this pull request?

File: maths/sum_of_digits.py [hunk 1]
```diff
@@ -55,9 +55,32 @@ def sum_of_digits_compact(n: int) -> int:
...
+    >>> digital_root(0)
+    0
+    >>> digital_root("109820394")
+    'Please enter a non-negative integer'
+    """
+
+    try:
+      return n % 9 or n and 9 if n >= 0 else "Input wasn't valid"
+    except TypeError:
+      return "Please enter a non-negative integer"
```
Comment (poyea) [inline maths/sum_of_digits.py:78]:
```suggestion
    try:
        return n % 9 or n and 9 if n >= 0 else "Input wasn't valid"
    except TypeError:
        return "Please enter a non-negative integer"
```

Comment (meg-1) [reply maths/sum_of_digits.py]: > 

@poyea , thank you for your suggestion! just commited it.

File: maths/sum_of_digits.py [hunk 2]
```diff
@@ -85,6 +108,14 @@ def benchmark() -> None:
...
         "seconds",
     )
+    print(
+        "> digital_root()",
+        "\tans =",
+        digital_root(small_num),
+        "\ttime =",
+        timeit("z.digital_root(z.small_num)", setup="import __main__ as z"),
+        "seconds",
+    )
```
Comment (cclauss) [inline maths/sum_of_digits.py:118]:
```suggestion
    result = timeit("z.digital_root(z.small_num)", setup="import __main__ as z")
    print(f"> digital_root()\tans = {digital_root(small_num)}\ttime = {result} seconds")
```

Comment (meg-1) [reply maths/sum_of_digits.py]: @cclauss , thank you for adding the enhancement!

**Comment by meg-1** [comment]
@poyea , hi! i would like to ask if the error shown after running my file in this commit is on my side, or on the repository's??
image
here's an attatched screen.

**Comment by cclauss** [comment]
Please rebase this pull request.

**Pull request closed without merging**